assayist/processor/base.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from abc import ABC, abstractmethod
+import json
+import neomodel
+import os
+
+from assayist.common.models import content, source
+from assayist.processor.configuration import config
+from assayist.processor.logging import log
+
+
+class Analyzer(ABC):
+    """Base Abstract class that analyzers will inherit from."""
+
+    METADATA_DIR = '/metadata/'
+    MESSAGE_FILE = METADATA_DIR + 'message.json'
+    BUILD_FILE = METADATA_DIR + 'buildinfo.json'
+    TASK_FILE = METADATA_DIR + 'taskinfo.json'
+    RPM_FILE = METADATA_DIR + 'rpms.json'
+    ARCHIVE_FILE = METADATA_DIR + 'archives.json'
+    IMAGE_RPM_FILE = METADATA_DIR + 'image-rpms.json'
+    BUILDROOT_FILE = METADATA_DIR + 'buildroot-components.json'
+
+    def main(self):
+        """Call this to run the analyzer."""
+        neomodel.config.DATABASE_URL = config.DATABASE_URL
+        neomodel.config.AUTO_INSTALL_LABELS = True
+        # run the analyzer in a transaction
+        neomodel.db.begin()
+        try:
+            self.run()
+            log.debug("Analyzer completed successfully, committing.")
+            neomodel.db.commit()
+        except Exception as e:
+            log.exception("Error encountered executing Analyzer, rolling back transaction.")
+            neomodel.db.rollback()
+            raise
+
+    @abstractmethod
+    def run(self):
+        """Implement anlyzer code here in your subclass."""
+
+    def read_metadata_file(self, FILE):
+        """Read and return the specified json metadata file or an empty dict."""
+        if os.path.isfile(FILE):
```

Comment:
I'd say we should document the inconsistency. We'll have normal conditions where `archives.json` is missing. I suppose we could rely on the analyzer to detect these conditions and avoid asking for that file, but IMO returning an empty result here makes sense too. It might even make the logic simpler, especially for imported builds (where `taskinfo.json` will be missing).